app/concerns/agent_controller_concern.rb

```diff
@@ -68,7 +68,8 @@ def control!
             log "Agent '#{target.name}' is disabled"
           end
         when 'configure'
-          target.update! options: target.options.deep_merge(interpolated['configure_options'])
+          new_options = target.options.deep_merge(interpolated['configure_options']) { |key, old, new| !(new.kind_of?(Array) && old.kind_of?(Array)) ? new : (old+new).uniq }
```

Comment:
That functionality is actually what is being replaced in this pr. I tend to favor non-destructive logic, i.e. merge > replace. 

For context, I'm trying to configure multiple similar agents using the commander agent. Each agent being commanded may have options unique to itself. 

When there are Arrays within the options hash:

...Defaulting to replace-logic deletes commanded-agent-specific options.

... Defaulting to Merge-logic doesn't remove options that are supposed to be replaced.

The simplest solution I've come up with is updating the[, in this case, Trigger] agent to accept an array-less options hash.

It'd be nice to somehow advise new developers that storing options objects within an array may have unforeseen consequences.